commandapi-platforms/commandapi-spigot/commandapi-spigot-core/src/main/java/dev/jorel/commandapi/CommandAPISpigot.java

```diff
@@ -0,0 +1,137 @@
+package dev.jorel.commandapi;
+
+import dev.jorel.commandapi.commandsenders.BukkitBlockCommandSender;
+import dev.jorel.commandapi.commandsenders.BukkitCommandSender;
+import dev.jorel.commandapi.commandsenders.BukkitConsoleCommandSender;
+import dev.jorel.commandapi.commandsenders.BukkitEntity;
+import dev.jorel.commandapi.commandsenders.BukkitNativeProxyCommandSender;
+import dev.jorel.commandapi.commandsenders.BukkitPlayer;
+import dev.jorel.commandapi.commandsenders.BukkitProxiedCommandSender;
+import dev.jorel.commandapi.commandsenders.BukkitRemoteConsoleCommandSender;
+import dev.jorel.commandapi.exceptions.WrapperCommandSyntaxException;
+import dev.jorel.commandapi.nms.NMS;
+import dev.jorel.commandapi.nms.SpigotNMS;
+import dev.jorel.commandapi.wrappers.NativeProxyCommandSender;
+import net.md_5.bungee.api.chat.BaseComponent;
+import org.bukkit.Bukkit;
+import org.bukkit.command.BlockCommandSender;
+import org.bukkit.command.CommandMap;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.ConsoleCommandSender;
+import org.bukkit.command.ProxiedCommandSender;
+import org.bukkit.command.RemoteConsoleCommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.server.ServerLoadEvent;
+import org.bukkit.plugin.java.JavaPlugin;
+
+public abstract class CommandAPISpigot<Source> extends CommandAPIBukkit<Source> implements SpigotNMS<Source> {
+
+	private static CommandAPISpigot<?> spigot;
+
+	@SuppressWarnings("unchecked")
+	protected CommandAPISpigot() {
+		this.nms = (NMS<Source>) bukkitNMS();
+		CommandAPISpigot.spigot = this;
+		setInstance(this);
+	}
+
+	@SuppressWarnings("unchecked")
+	public static <Source> CommandAPISpigot<Source> getSpigot() {
+		return (CommandAPISpigot<Source>) spigot;
+	}
+
+	public static InternalSpigotConfig getConfiguration() {
+		return (InternalSpigotConfig) CommandAPIBukkit.getConfiguration();
+	}
+
+	private static void setInternalConfig(InternalSpigotConfig config) {
+		CommandAPIBukkit.config = config;
+	}
+
+	@Override
+	public <T extends CommandAPIBukkitConfig<T>> void onLoad(CommandAPIBukkitConfig<T> config) {
+		if (config instanceof CommandAPISpigotConfig spigotConfig) {
+			CommandAPISpigot.setInternalConfig(new InternalSpigotConfig(spigotConfig));
+		} else {
+			CommandAPI.logError("CommandAPIBukkit was loaded with non-Bukkit config!");
+			CommandAPI.logError("Attempts to access Bukkit-specific config variables will fail!");
+		}
+		onLoad();
```

Comment:
It would be clearer this calls the method in `CommandAPIBukkit` if it used the `super` keyword. At the moment, I was like, "Wait, where's the no-parameter `onLoad` method in `CommandAPISpigot`?" when I saw this.

```suggestion
		super.onLoad();
```